### ISSUE TITLE
Material assignment problems

### ISSUE THREAD

**Issue by gabrielmontagne** [makehumancommunity/mpfb2#7]
I haven't come up with a reliable way to reproduce this -- or to avoid this!  But depending on the order in which I apply my assets, sometimes they end up getting the eye material!

I've seen it with eyebrows, eyelashes, etc.   Today is the first time I see the tongue. 

- ![](https://firebasestorage.googleapis.com/v0/b/firescript-577a2.appspot.com/o/imgs%2Fapp%2Fyqm%2FBoLBlbsacH.png?alt=media&token=472de78b-aac5-4f49-bcfb-b73fbb5af674)

**Comment by joepal1976** [comment]
Hm. Yes, I've seen that once or twice too. I had concluded I was crazy or did something wrong. 

Can you set the debug level to "DUMP" for the following channels, and then attach a log if you see it happen again?

- services.clothesservice
- services.humanservice
- services.materialservice
- services.nodeservice
- assetlibrary.loadlibraryclothes
-

**Comment by gabrielmontagne** [comment]
Sorry, I could find how to set the debug level to more than 2 channels -- I set it generally to `DUMP` and, well here it comes from a `tee` in all its glory.

To reproduce is easy: create the eyes first, then apply the other assets.

[mpfb2.log](https://github.com/makehumancommunity/mpfb2/files/6610850/mpfb2.log)

I hope you won't get nightmares from the following screenshot!

![feh_467423_000001_2021-06-07-190804_1028x646_scrot](https://user-images.githubusercontent.com/95913/121061160-ce61d400-c7c3-11eb-8154-46a03d05219e.png)

**Comment by gabrielmontagne** [comment]
Now I see how debugging works!  Let me try again

**Comment by gabrielmontagne** [comment]
Here the correct logs,

[run.txt](https://github.com/makehumancommunity/mpfb2/files/6610926/run.txt)

Thanks for looking into this!

**Comment by joepal1976** [comment]
Thanks. I think https://github.com/makehumancommunity/mpfb2/commit/8e04bc47d5c0da129d0729b2a3c56f0513ee02d0 will solve the issue. If you run from source you could test it now, otherwise it'll end up in tomorrow's build. 

While the fix seem to work, I still don't see why it didn't work before. There's something I don't understand with how the eye material setting could leak from one operator instance to another. I'll keep a suspicious watch over these areas in case similar problems arise later on.

**Comment by gabrielmontagne** [comment]
That's great :-)  I running the plugin from the commit and I cannot reproduce anymore!  That was quick!
